src/main/java/org/passay/dictionary/BloomFilterDictionary.java

```diff
@@ -0,0 +1,77 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay.dictionary;
+
+import com.google.common.hash.BloomFilter;
+
+/**
+ * Dictionary that is backed by a <a href="https://en.wikipedia.org/wiki/Bloom_filter">Bloom Filter</a>.
+ * <strong>WARNING</strong> bloom filters may return true for a word that is <strong>NOT</strong> in the dictionary.
+ * This implementation should only be used if false positives can be tolerated.
+ *
+ * @author  Middleware Services
+ */
+public class BloomFilterDictionary implements Dictionary
+{
+
+  /** Filter used for searching. */
+  private final BloomFilter<String> bloomFilter;
+
+
+  /**
+   * Creates a new dictionary instance from the supplied {@link BloomFilter}.
```

Comment:
You should consider making a recommendation here or in the class javadoc for the false-positive probability, or at least discuss tradeoffs. I'm concerned that the default Guava value of 3% is too high.

Reply:
I'm not sure what I'd say. I've tried to make it clear that bloom filters will operate differently than the other dictionaries. But advice about a particular fpp seems very subjective.

Reply:
In the spirit of being descriptive over prescriptive, I'll take a stab at a helpful note to be added to what you have for the constructor:

> The _fpp_ (false-positive probability) parameter of the given Bloom filter is a vitally important configuration concern. If it is too high, one risks user frustration due to rejection of valid passwords; if it is too low, one risks excessive storage costs. Finding the proper balance between acceptable user experience and storage costs is worth the time and effort required in testing. The Guava default value of 3% is likely unsuitable for many if not most deployments.

Reply:
Javadocs updated.